test/nodejs/table.js

```diff
@@ -6,178 +6,187 @@ var three_test_coordinates = require('./constants').three_test_coordinates;
+            coordinates: [three_test_coordinates[0], three_test_coordinates[1]],
+            annotations: [annotation.slice(0,-1)]
+        };
+        osrm.table(options, function(err, table) {
+            assert.ifError(err);
+            assert.ok(Array.isArray(table[annotation]), 'result must be an array');
+            var row_count = table[annotation].length;
+            for (var i = 0; i < row_count; ++i) {
+                var column = table[annotation][i];
+                var column_count = column.length;
+                assert.equal(row_count, column_count);
+                for (var j = 0; j < column_count; ++j) {
+                    if (i == j) {
+                        // check that diagonal is zero
+                        assert.equal(0, column[j], 'diagonal must be zero');
+                    } else {
+                        // everything else is non-zero
+                        assert.notEqual(0, column[j], 'other entries must be non-zero');
+                        // and finite (not nan, inf etc.)
+                        assert.ok(Number.isFinite(column[j]), 'distance is finite number');
+                    }
                 }
             }
-        }
-        assert.equal(options.coordinates.length, row_count);
+            assert.equal(options.coordinates.length, row_count);
+        });
     });
-});
 
-test('table: distance table in Monaco with sources/destinations', function(assert) {
-    assert.plan(7);
-    var osrm = new OSRM(data_path);
-    var options = {
-        coordinates: [three_test_coordinates[0], three_test_coordinates[1]],
-        sources: [0],
-        destinations: [0,1]
-    };
-    osrm.table(options, function(err, table) {
-        assert.ifError(err);
-        assert.ok(Array.isArray(table.durations), 'result must be an array');
-        var row_count = table.durations.length;
-        for (var i = 0; i < row_count; ++i) {
-            var column = table.durations[i];
-            var column_count = column.length;
-            assert.equal(options.destinations.length, column_count);
-            for (var j = 0; j < column_count; ++j) {
-                if (i == j) {
-                    // check that diagonal is zero
-                    assert.equal(0, column[j], 'diagonal must be zero');
-                } else {
-                    // everything else is non-zero
-                    assert.notEqual(0, column[j], 'other entries must be non-zero');
-                    // and finite (not nan, inf etc.)
-                    assert.ok(Number.isFinite(column[j]), 'distance is finite number');
+    test('table: ' + annotation + ' table in Monaco with sources/destinations', function(assert) {
+        assert.plan(7);
+        var osrm = new OSRM(data_path);
+        var options = {
+            coordinates: [three_test_coordinates[0], three_test_coordinates[1]],
+            sources: [0],
+            destinations: [0,1],
+            annotations: [annotation.slice(0,-1)]
```

Comment:
The fact that our annotations parameters uses singular over plural is one of my biggest API regrets. :see_no_evil:

Reply:
😢